### PR TITLE
Move Bank into solana_runtime crate

### PR DESCRIPTION
**Pull request by garious**
#### Problem

This is what smart contract tests look like using the low-level library `runtime`:
https://github.com/solana-labs/solana/blob/master/programs/native/rewards/tests/rewards.rs#L13

This is what smart contract tests look like using the high-level library `bank`:
https://github.com/solana-labs/solana/blob/master/src/bank.rs#L692

#### Summary of Changes

Moved `bank.rs` and its dependencies into the `solana_runtime` crate.

sub-second unit-test suite!

### CODE REVIEW

File: runtime/src/accounts.rs [hunk 1]
```diff
@@ -256,13 +255,13 @@ impl AccountsDB {
...
         self.transaction_count
     }
-    pub fn account_values_slow(&self) -> Vec<(Pubkey, solana_sdk::account::Account)> {
-        self.accounts.iter().map(|(x, y)| (*x, y.clone())).collect()
-    }
-    fn merge(&mut self, other: Self) {
-        self.transaction_count += other.transaction_count;
-        self.accounts.extend(other.accounts)
-    }
+    //pub fn account_values_slow(&self) -> Vec<(Pubkey, solana_sdk::account::Account)> {
```
Comment (garious) [inline runtime/src/accounts.rs]: @aeyakovenko, https://github.com/solana-labs/solana/blob/master/CONTRIBUTING.md#should-i-send-small-prs-as-i-develop-large-new-components

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -1,209 +1,10 @@
...
-    #[should_panic]
-    fn test_get_subset_unchecked_mut_out_of_bounds() {
-        // This panics, because it assumes bounds validation is done elsewhere.
-        get_subset_unchecked_mut(&mut [7, 8], &[2]);
-    }
-}
+mod accounts;
+pub mod bank;
+pub mod bloom;
+pub mod genesis_block;
```
Comment (garious) [inline runtime/src/lib.rs]: @mvines, want this in `solana_runtime` or `solana_sdk`?  I'm thinking SDK might be better.

**Comment by garious** [comment]
@mvines, remind me how we upgrade Rust on the stable-perf machines?

File: runtime/src/lib.rs (hunk 2 above)
Comment (mvines) [reply runtime/src/lib.rs]: ☑️ sdk

**Pull request merged**